### DISCUSSION TITLE
Data extrapolation, forecast low and forecast low alert doesn't work

### DISCUSSION THREAD

**Discussion by Morton80** [NightscoutFoundation/xDrip#2972] [Q&A]
Hi there!
I am using xdrip since several years and it significantly improved my life. But some months ago, the dotted BG extrapolation curve and low prediction time disappeared without any known reason. Consequently, the forecast low alert doesn't work anymore since then. I checked all settings, re-installed older an new xdrip versions, nothing helped. Yesterday I changed to a new phone, installed xdrip from scratch with Libre 2 direct as data source, and I didn't copy the settings from my old phone. The extrapolation curve was then visible for one hour and disappeared again. The only thing I did in between was pairing my smart watch and installing xdrip on the watch.
On the old phone (Galaxy A12 with Android 12), I was using Libre 2 with the patched app first and Libre bluetooth as data source lateron, but change from patched app to Libre bluetooth with OOP2 didn't have an effect on the low prediction (as expected).
All other alarms work properly.
My new phone now is a Galaxy A54 5G with Android 13.
I hope anybody can help since I'm quite desperate now... 😕 
Thanks in advance!
![Screenshot_20230727_140226_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/275e106a-1ccf-4829-83bb-a4db280c0447)
![Screenshot_20230727_113147_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/40c87b33-0b40-4fb0-9a34-8d47e75b29f4)
![Screenshot_20230726_223241_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/0dea5669-abff-4889-9837-bb10a341937d)
![Screenshot_20230726_223213_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/889b27b8-6b5c-4586-8d14-0e16c7354496)

**Comment by Navid200** [comment]
I'm not 100% sure this would/could be the cause:
Is it possible that you have disabled Momentum?


Settings &#8722;> xDrip+ Predictive Simulation settings &#8722;> Use trend momentum

**Reply by Morton80** [reply]
No, it's enabled... Doesn't seem to be the reason. 

Are there any other setting pages I could share?

**Reply by Navid200** [reply]
@jamorham What could cause the extrapolation to stop working when the setting is still enabled?

**Comment by Navid200** [comment]
So, you never get the red dashed extrapolation line anymore?

**Reply by Morton80** [reply]
Yes, correct. Still no extrapolation line and no forecast low alert.

**Comment by Navid200** [comment]
Looking at your main screen screenshot, it looks like there are missing points.  The forecast low alert needs consistent readings.
I wonder if that is the cause.  And I don't know why you don't have consistent readings.

**Reply by Morton80** [reply]
Thank you for coming back to my question.
For me, it seems that the readings are complete (blue dots), but only the gray intrapolation line has some gaps - for whatever reason.
I have observed in the meantime, that the red extrapolation curve and also the low prediction works once in a while, maybe every two days for a few minutes, and disapears then again without any clear reason. So the values are still falling, but suddenly no extrapolation anymore...

**Comment by Navid200** [comment]
This is what I expect to see, 12 readings every 1 hour since 12 x 5min = 60min.
![Screenshot_20231010-104259](https://github.com/NightscoutFoundation/xDrip/assets/51497406/468e0f46-5dc1-4530-a64d-fcbe948c00ec)
  
But, this is what I see when I look at your screen:
![256531889-275e106a-1ccf-4829-83bb-a4db280c0447](https://github.com/NightscoutFoundation/xDrip/assets/51497406/be4daba6-9507-4788-a4eb-6a204ec3ac82)
  
Do you see the difference?  
I don't understand why your readings are spaced like that.

**Reply by Morton80** [reply]
Yes, I see the difference and you are completely right. 
It seems to be some kind of time synchronisation issue between sensor,  OOP2 app and xdrip.
I checked the log and see that I receive every 2 minutes a datapoint from OOP algorithm:
![Screenshot_20231011_071152_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/970234d5-9698-4061-bdcc-0202ab9f7381)

Maybe this is problem. 2 min doesn't fit with 5 min.

**Comment by Navid200** [comment]
@psonnera
What could cause Libre readings to be spaced at a timing gap greater than 5 minutes?

**Reply by psonnera** [reply]
@Morton80 did you try to enable 1 minute reading by any chance? I don't know why OOP2 triggers every 2 minutes, it usually only does that every 5 minutes as xDrip+ leads the game in getting data from Libre 2. Libre 2 will send a reading every minute.

**Reply by Morton80** [reply]
The Libre one minute option is only applicable for patched app data source, correct? 
I have now enabled the Libre trend curve, which shows frequent dots:
![Screenshot_20231012_231612_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/22cff62f-cfeb-4559-829c-dea7e07d7920)

**Reply by Navid200** [reply]
But, you still don't have (non-raw) data once every 5 minutes.

I am not 100% sure.  But, I believe there is a part of the forecast low alert that confirms the forecast will be reliable before it becomes available and it expects readings once every 5 minutes if I'm not mistaken.

**Reply by psonnera** [reply]
@Morton80 1 minute is for direct connection too

**Reply by Morton80** [reply]
@Navid200  yes, I think this makes sense.
@psonnera  I managed to enable 1 minute reading, but still, just 10 readings per hour... 
Phone reboot was done. 

![Screenshot_20231014_202148_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/140705705/3e68a8b2-c093-4a1f-b178-ee614b2dfbbe)
